core/src/graph/mod.rs

```diff
@@ -173,8 +175,9 @@ where
     })
 }
 
+
 derive_from! {
-    #[derive(Debug, Clone)]
+    #[derive(Debug, Clone, Serialize)]
```

Comment:
Where do we need to serialize this?

Reply:
It is used for the implementation of `Display` (which calls `serde_json::to_string`)